baseapp/testutil/messages.go

```diff
@@ -0,0 +1,63 @@
+package testutil
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+)
+
+func RegisterInterfaces(registry types.InterfaceRegistry) {
+	registry.RegisterImplementations(
+		(*sdk.Msg)(nil),
+		&MsgCounter{},
+		&MsgCounter2{},
+		&MsgKeyValue{},
+	)
+	msgservice.RegisterMsgServiceDesc(registry, &_Counter_serviceDesc)
+	msgservice.RegisterMsgServiceDesc(registry, &_Counter2_serviceDesc)
+	msgservice.RegisterMsgServiceDesc(registry, &_KeyValue_serviceDesc)
+
+	codec.RegisterInterfaces(registry)
+}
+
+var _ sdk.Msg = &MsgCounter{}
+
+func (msg *MsgCounter) GetSigners() []sdk.AccAddress { return []sdk.AccAddress{} }
+func (msg *MsgCounter) ValidateBasic() error {
+	if msg.Counter >= 0 {
+		return nil
+	}
```

Comment:
Is it not possible to have zero messages?

Reply:
zero messages?

Reply:
I was reading this wrong, this makes sense